build/maintainer-flags.txt

```diff
@@ -10,7 +10,6 @@
 -Wno-strict-aliasing
 -Wno-uninitialized
 -Wredundant-decls
--Wreturn-type
```

Comment:
I'm curious as to why we're removing this one, it seems to fall into the "almost always an error" category?

Reply:
It was moved into the `MongoC-Warnings.cmake` file. This just removes the redundancy.